src/assignment3/Assignment3.java

```diff
@@ -0,0 +1,67 @@
+package assignment3;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class Assignment3 {
+
+    private static void recursiveSearch(char[][] grid, boolean visited[][],

```

Comment:
Generally, giving generic names to methods, like `recursiveSearch`, makes your code harder to read, even if they are helper methods. Maybe you could consider renaming this method.